libsolidity/parsing/Token.cpp

```diff
@@ -80,8 +141,70 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-Token::Value Token::fromIdentifierOrKeyword(const std::string& _name)
+unsigned Token::extractM(string const& _literal)
+{
+	try
+	{
+		unsigned short m = stoi(_literal.substr(_literal.find_first_of("0123456789")));
+		return m;
+	}
+	catch(out_of_range& e)
+	{
+		return 0;
+	}
+}
+pair<unsigned, unsigned> Token::extractMxN(string const& _literal)
+{
+	try
+	{
+		unsigned short m = stoi(_literal.substr(0, _literal.find_last_of("x") - 1));
```

Comment:
This should never be called if there is no x in the literal. It's only use is for FixedMxN...nothing else.